docs/interactions/Application_Commands.md

```diff
@@ -883,15 +894,93 @@ An autocomplete interaction **can return partial data** for option values. Your
 }
 ```
 
+## Localization
+
+Slash commands may be localized, which will cause them to use localized names and descriptions depending on the client's selected language.  This is entirely optional.  Localization is available for names and descriptions of commands, subcommands, and options, as well as the names of choices, by submitting the appropriate `name_localizations` and `description_localizations` fields when creating or updating the application command.
+
+Application commands may be partially localized - not all [available locales](#DOCS_REFERENCE/locales) are required, nor do different fields within a command need to support the same set of locales.  If a locale is not present in a localizations dictionary for a field, users in that locale will see the default value for that field.  It's not necessary to fill out all locales with the default value.  Any localized values that are identical to the default will be ignored.
+
+Localized option names are subject to an additional constraint, which is that they must be distinct from all other default option names of that command, as well as all other option names within that locale on that command.
+
+When taking advantage of slash command localization, the interaction payload received by your client will still use default command, subcommand, and option names.  To localize your interaction response, you can determine the client's selected language by using the `locale` key in the interaction payload.
+
+An application command furnished with localizations might look like this:
+
+```json
+{
+  "name": "birthday",
+  "type": 1,
+  "description": "Wish a friend a happy birthday",
+  "name_localizations": {
+    "zh-CN": "生日",
+    "el": "γενέθλια"
+  },
+  "description_localizations": {
+    "zh-CN": "祝你朋友生日快乐"
+  },
+  "options": [
+    {
+      "name": "age",
+      "type": 4,
+      "description": "Your friend's age",
+      "name_localizations": {
+        "zh-CN": "岁数"
+      },
+      "description_localizations": {
+        "zh-CN": "你朋友的岁数"
+      }
+    }
+  ]
+}
+```
+
+### Retrieving localized commands
+
+While most endpoints that return application command objects will return `name_localizations` and `description_localizations` fields, some will not.  This includes `GET` endpoints that return all of an application's guild or global commands.  Instead, those endpoints will supply additional `name_localized` or `description_localized` fields, which contain only the localization relevant to the requester's locale.
```

Comment:
Unless I'm misreading/misunderstanding, doesn't this mess with diffing commands to determine whether to update some? Namely in the case a localization string was updated for the locale not requested.

Is there a reason this limit is in place?

Reply:
I don't think the primary use of this API is for management (in particular, the clients may use it to discover application commands), so it's important for it to be relatively lean by default.  You can supply `with_localizations` as a query argument if you need the entire dictionary.  I'll make it so supplying that argument disables the `name_localized` and `description_localized` fields, so that raw diffing will work.

Reply:
Aha I missed that parameter -- could that be mentioned here for the next person like me? 😅